source/components/composing-components.md

```diff
@@ -0,0 +1,260 @@
+
+```app/templates/application.hbs
+{{#user-list users=activeUsers sortBy='name' as |user|}}
+  {{user-card user=user}}
+{{/user-list}}
+```
+
+### Component Blocks
+
+Components can be used in two forms, just like regular HTML elements.
+
+**Inline Form**
+
+```app/templates/application.hbs
+{{user-list users=activeUsers}}
+```
+
+This is the basic blockless form of a component, which doesn't give the
+consumer the ability to insert their own content into the given component.
+It's even possible that a component can only be used in this form.
+Make sure to consult a component's documentation for its available uses
+and forms.
+
+**Block Form**
+
+```app/templates/application.hbs
+{{#user-list users=activeUsers}}
+  {{!-- custom template here --}}
+{{/user-list}}
+```
+
+To compose components, we must use the block form, but we must also
+be able to distinguish from within our component which form is being used.
+This can be done with the `hasBlock` property from within our component.
+
+```app/templates/components/user-list.hbs
+{{#if hasBlock}}
+  {{yield}}
+{{else}}
+  <p>No Block Specified</p>
+{{/if}}
+```
+
+Once we know that the component is being used in block form, we can yield
+whatever the user has placed inside the block wherever we want via the `{{yield}}` helper.
+
+This `{{yield}}` helper can be used in the component's template definition multiple times,
+allowing us to create a component that works like a list, where the user insert/modify custom content
+at every yield. Take the following example, where we output a custom summary.
+
+```app/templates/components/post-list.hbs
+{{#each posts as |post|}}
+  <h3>{{post.title}}</h3>
+  <p>{{yield}}</p>
+{{/each}}
+```
+
+Which can be used like so:
+
+```app/templates/posts.hbs
+{{#post-list posts=headlinePosts}}
+  Greatest post ever!
+{{/post-list}}
+```
+
+And will result in the following HTML:
+
+```html
+<div id="ember123" class="ember-view">
+  <h3>Tomster goes to town</h3>
+  <p>Greatest post ever!</p>
+  <h3>Tomster on vacation</h3>
+  <p>Greatest post ever!</p>
+</div>
+```
+
+But what use is it to just output the same thing over and over? Don't we want to customize our posts,
+and display the right content? Sure we do. Lets explore the `{{yield}}` helper a bit.
+
+### Data Down
+
+To accomplish composability beyond just simple templates, we need to pass context to those templates.
+This can be done with the `{{yield}}` helper that was just introduced above.
+
+The `{{yield}}` defines where the content we defined inside our component block will yield in the component's layout,
+as we saw in the previous section. Apart from that, the yield helper also allows us to send data down, providing a context for
+the content inside of the component's block. This allows anyone using our component to get context about what's going on
+inside, and to customize the output of the component with that information.
```

Comment:
That phrase `context` might be better as `params` here. `context` of a template means something fairly specific, the `this` of a template. We definitely call it `context` elsewhere.

Starting to introduce "block params" by talking about `yielding params back to the scope the component was invoked in` seems good here.